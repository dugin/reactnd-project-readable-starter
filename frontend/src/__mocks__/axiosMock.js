import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import {categories, posts} from "./responseMock";

export const mock = new MockAdapter(axios);

export const URL = 'http://localhost:3001';

mock.onGet(URL + '/categories').reply(200, categories);

mock.onGet(URL + '/posts').reply(200, posts);


