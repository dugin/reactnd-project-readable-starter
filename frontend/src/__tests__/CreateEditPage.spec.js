import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {CreateEditPage} from "../components/create-edit-page/CreateEdit.page";
import {mock, URL} from '../__mocks__/axiosMock';

describe('CreateEditPage Component', () => {

    const props = {history: {goBack: jest.fn()}};

    let createEditPage;

    beforeEach(() => {
      //  mock.onPost(URL + '/posts').reply(200);

        createEditPage = shallow(<CreateEditPage {...props}/>);
    });


    it('renders properly', () => {

        expect(toJson(createEditPage)).toMatchSnapshot();

    });

    it('on submit', () => {
        createEditPage.instance().setPost = jest.fn();

        createEditPage.find('form')
            .simulate('submit', {
                preventDefault() {
                }
            });

        expect(createEditPage.instance().setPost).toHaveBeenCalled();
    });

    it('on addPost', async () => {

        mock.onPost(URL + '/posts').reply(200);

        await createEditPage.instance().setPost();

        expect(createEditPage.instance().props.history.goBack).toHaveBeenCalled();
    });

    it('on change category', () => {

        createEditPage.instance().onCategory('test');

        expect(createEditPage.state().category).toEqual('test');
    });



});

describe('onError Component', () => {
    mock.onPost(URL + '/posts').reply(404);

    const props = {history: {goBack: jest.fn()}};

    let createEditPage;

    beforeEach(() => {
        createEditPage = shallow(<CreateEditPage {...props}/>);


    });
    it('on post error', async () => {

        await createEditPage.instance().setPost();

        expect(createEditPage.state().error).toExist();
    });
})
