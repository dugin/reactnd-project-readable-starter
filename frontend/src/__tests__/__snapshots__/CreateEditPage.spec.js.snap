// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateEditPage Component renders properly 1`] = `
<form
  className="CreateEditPage mt-3 row"
  onSubmit={[Function]}
>
  <div
    className="form-group col-12 "
  >
    <input
      className="form-control"
      name="title"
      placeholder="Title"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-group col-12 "
  >
    <textarea
      className="form-control"
      name="body"
      required={true}
      rows="6"
    />
  </div>
  <div
    className="row mx-0"
  >
    <div
      className="form-group col-7"
    >
      <input
        className="form-control"
        name="author"
        placeholder="Author"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-group col-5 pl-0"
    >
      <Categories
        onEdit={true}
        selected={[Function]}
      />
    </div>
  </div>
  <div
    className="col-12 CreateEditPage__submit"
  >
    <p
      className="required-field"
    >
      * All fields are required
    </p>
    <button
      className="btn btn-success btn-block"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
